src/app/WriteHandler.cpp

```diff
@@ -110,6 +110,10 @@ CHIP_ERROR WriteHandler::SendWriteResponse()
 CHIP_ERROR WriteHandler::ProcessAttributeDataIBs(TLV::TLVReader & aAttributeDataIBsReader)
 {
     CHIP_ERROR err = CHIP_NO_ERROR;
+
+    // TODO: does exchange context always have session handle? what if it doesn't?
```

Comment:
It's hard to imagine how this could happen.  But I think if it does, it means we just have to disallow everything.

If in the future we decide to store some of the subject descriptor information in the underlying secure session instead of the session handle, we'll have to make similar choices if we encounter a dangling session handle for which the underlying session has been evicted.

That also seems unlikely, but would probably still need some handling in the code.